### ISSUE TITLE
Tiny MCE editor messes up our URL

### ISSUE THREAD

**Issue by bethlakshmi** [bethlakshmi/gbe-divio-djangocms-python2.7#85]
Finally replicated on Firefox on my machine, but happens on Chrome AND Firefox on @burlexpo's.

Test that definitely fails = edit email template, make a link, close the link box, reopen it - and it shows "../.." instead of "http://buresque-expo.com".

**Comment by bethlakshmi** [comment]
Having trouble replicating this with bulk edit.

**Comment by bethlakshmi** [comment]
Please test:
https://burlesque-expo-stage.us.aldryn.io/email/list_template

I think I found a fix - I ripped out the django package that gave me TinyMCE and integrated it directly into the template.  That works fine with the rest of the code, and gives me more control. 

Biggest thing - I upgraded us 2 version from Tiny MCE 3 to 5.
Smaller thing - I have more direct control over it's config settings, so I was able to do some better tuning of the UI - a nice little tidbit is I put two of your most common links (the main page, and the user home page, as links you can insert.

Tested with my Firefox and problem is solved.  But given how browser dependant this is - I want confirmation.

If that works, I'll go forth and update all the other pages using TinyMCE and rip out the old way completely.

**Comment by bethlakshmi** [comment]
OK!  Last night I figured out a better answer to the issue with empty fields and TinyMCE - for posterity... it turns out that TinyMCE and the HTML5 "required" field don't play well together in the latest version of Tiny MCE.  Our really old version did it's thing a different way, and so the two systems didn't conflict.  But now, with the way Tiny MCE works, it's a problem.  That is at the core of why I can't get the submit buttons to work when the text area field was initialized with no data.  

That's a nicer thing to fix - Django 1.10 (we use 1.11) introduced the use of the "required" field by default in forms. Most of the time, it's really nice - it lets the form bug the user via the browser with popups to fill in data before sending the data to the server - that's a nicer user experience, as it's less wait time (and less load on the server).  But ... clearly right now it's generating a difficult type of evil.

BUT - Django will let me turn off the publication of "required" as something rendered in the HTML of a form, while still making the form treat that field as required when it gets processed in the server - so the user will still get an error message, but the error message comes from Django code, not the browser..... and doesn't interfere with Tiny MCE!!  Ever!!

Fixed all the forms relying on Tiny MCE data.  Did a pull request.  Rerunning test coverage now (the change in "required" broke 2 tests).  But largely done here.
